src/ServiceManagement/Compute/Commands.ServiceManagement.Preview/Automation/RunCodeGeneration.ps1

```diff
@@ -198,6 +199,56 @@ function Get-NormalizedTypeName
     return $outputName;
 }
 
+function Is-SpecialType
+{
+    param(
+        [Parameter(Mandatory = $True)]
+        [string]$inputName
+    )
+
+    return ($inputName.StartsWith($client_model_namespace_prefix));
+}
+
+function Is-ListType
+{
+    param(
+        [Parameter(Mandatory = $True)]
+        [System.Reflection.PropertyInfo[]]$property_info_array
+    )
+
+    if ($property_info_array.Count -eq 1)
+    {
+         if ($property_info_array[0].PropertyType.FullName -like '*List*System.String*')
```

Comment:
I think it's better to be more specific in determining the list and element types.